### PR TITLE
Improve test running display with test count and filenames

### PR DESCRIPTION
**Pull request by lindycoder**
Added an header, footer and the number of test that run
Tests are now stripped of the "test_" prefix along with the
filename in which they are located

A new concept has been introduced: the REGISTRY.
This is named after ASM registries and serve as a standard
registry of values for the tests to write in that the test runner
can read and interpret. The first of these value is the test count
that just ran.  Future addition will probably be failures, mock
expectations and such things that should be global to the
test run

### CODE REVIEW

File: src/test-running.sh [hunk 1]
```diff
@@ -25,6 +61,7 @@ _setup_workspace() {
     cp -aR ${SOURCE_ROOT}/* ${workspace}/
 
     mocks="$(mktemp -d "${workspace}/mocks.XXXXXXXX")"
+    mocks="$(mktemp -d "${workspace}/mocks.XXXXXXXX")"
```
Comment (mat128) [inline src/test-running.sh]: Duplicate line

File: src/test-running.sh [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
+  echo ""
+  echo ">>> SUCCESS <<<"
+  echo ""
   exit 0
 fi
 
 
 SOURCE_ROOT=$1
 TEST_FILE=$2
+REGISTRY=$3
```
Comment (mat128) [inline src/test-running.sh]: Can't see any usage?

File: src/test-running.sh (hunk 1 above)
Comment (mat128) [inline src/test-running.sh]: Nevermind ;)

**Comment by mat128** [comment]
i      :)

File: src/test-running.sh (hunk 2 above)
Comment (lindycoder) [inline src/test-running.sh]: Yes at line 107

File: src/test-running.sh (hunk 1 above)
Comment (lindycoder) [inline src/test-running.sh]: will fix

**Pull request merged**